draft-ietf-jsonpath-base.md

```diff
@@ -1036,28 +1036,30 @@ The current item is selected if and only if the result is `true`.
 
 
 ~~~~ abnf
+boolean-expr     = logical-or-expr / logical-and-expr
+logical-or-expr  = logical-and-expr "||" boolean-expr ; disjunction
+                                                      ; binds less tightly than conjunction
+logical-and-expr = basic-filter /
+                   basic-filter "&&" logical-and-expr ; conjunction
+                                                      ; binds more tightly than disjunction
 
-
-boolean-expr = logical-expr
-logical-expr = ([neg-op] primary-expr) / logical-or-expr
-neg-op       = "!"                                  ; not operator
-primary-expr = "(" logical-or-expr ")"
-logical-or-expr = logical-and-expr *["||" logical-and-expr]
-logical-and-expr = comp-expr *["&&" comp-expr]
+basic-filter = comp-expr / paren-expr / ([neg-op] paren-expr)
+paren-expr   = "(" boolean-expr ")"                   ; parenthesized expression
```

Comment:
Eventually, we'll want other-type expressions here, such as `4*(5+6)`.

Reply:
I don't intend to include that in this PR, thanks.